src/main/java/io/github/genomicdatainfrastructure/daam/services/SubmitApplicationService.java

```diff
@@ -0,0 +1,47 @@
+// SPDX-FileCopyrightText: 2024 PNED G.I.E.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package io.github.genomicdatainfrastructure.daam.services;
+
+import static io.github.genomicdatainfrastructure.daam.security.PostAuthenticationFilter.USER_ID_CLAIM;
+import io.github.genomicdatainfrastructure.daam.remote.rems.api.RemsApplicationCommandApi;
+import io.github.genomicdatainfrastructure.daam.remote.rems.model.SubmitApplicationCommand;
+import io.quarkus.oidc.runtime.OidcJwtCallerPrincipal;
+import io.quarkus.security.identity.SecurityIdentity;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+import java.util.List;
+
+@ApplicationScoped
+public class SubmitApplicationService {
+
+    private final SecurityIdentity identity;
+    private final String remsApiKey;
+    private final RemsApplicationCommandApi remsApplicationCommandApi;
+
+    @Inject
+    public SubmitApplicationService(
+        @ConfigProperty(name = "quarkus.rest-client.rems_yaml.api-key") String remsApiKey,
+        SecurityIdentity identity,
+        @RestClient RemsApplicationCommandApi applicationsApi
+    ) {
+        this.remsApiKey = remsApiKey;
+        this.identity = identity;
+        this.remsApplicationCommandApi = applicationsApi;
+    }
+
+    public void submitApplication(String id) {
+        var principal = (OidcJwtCallerPrincipal) identity.getPrincipal();
```

Comment:
suggestion: Can you please move the resolution of `userId` to the controller/api class? 

`OidcJwtCallerPrincipal` is part of authentication/authorization layer, whereas use case services must be agnostic of this kind of implementation details - imagine we want to support multiple authentication mechanisms.

Can you also please fix `CreateApplicationService`? I just missed this problem there as well.

Reply:
I will do that in separate PR if thats okay with you

Reply:
Sure, that's why I approved. I did not close the conversation just to ensure you get to read it.